x/examples/mobileproxy-clib/demo/demo.c

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// -- WORK IN PROGRESS --
+
+#include <stdio.h>
+#include <bin/mobileproxy-clib.h>
+
+// Is there a way to import these? Or do we need to define them ourselves?
+typedef unsigned int StreamDialerPtr;
+typedef unsigned int ProxyPtr;
+
+int main()
+{
+  StreamDialerPtr *dialer;
+  ProxyPtr *proxy;
+
+  dialer = NewStreamDialerFromConfig("split:3");
+  proxy = RunProxy("127.0.0.1:1234", dialer); // Is this call blocking? Do we need to run it in another thread?
+
+  // TODO: Wait for terminate key, we may read from console using fgetc
+  // Signal (i.e. Ctrl+C) is a Unix/Linux only API, Windows doesn't use it
+  // So it also depends on whether we need the program to be cross-platform
+  printf("Running proxy on 127.0.0.1:1234. Press <Enter> to terminate: ");
+
```

Comment:
There's no call to read the Enter here.

Consider `sigaction`: https://stackoverflow.com/a/19059462

Reply:
`sigaction` is not cross-platform. I'd rather use a standard library function.

Reply:
What do you mean? It's in the POSIX standard: https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigaction.html

Does Windows not implement it? I guess signal is probably fine too.

Reply:
Not `sigaction`, but [`signal` is good](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-170).